lib/process_helpers.py

```diff
@@ -2,36 +2,26 @@
 import os
 import subprocess
 
-def kill_ps(ps_to_kill):
-    print('Killing processes')
-    for ps_info in ps_to_kill:
-        pid = ps_info['ps'].pid
-        print(f"Trying to kill {ps_info['cmd']} with PID: {pid}")
-        try:
-            if ps_info['ps_group'] is True:
-                try:
-                    ps_group_id = os.getpgid(pid)
-                    print(f" with process group {ps_group_id}")
-                    os.killpg(os.getpgid(pid), signal.SIGTERM)
-                    try:
-                        ps_info['ps'].wait(timeout=5)
-                    except subprocess.TimeoutExpired:
-                        # If the process hasn't gracefully exited after 5 seconds we kill it
-                        os.killpg(os.getpgid(pid), signal.SIGKILL)
-                except ProcessLookupError:
-                    # process may be not have been in a process group
-                    print(f"Could not find process-group for {pid}")
-            # always, just in case the calling process (typically the shell) did not die
-            os.kill(pid, signal.SIGTERM)
-            try:
-                ps_info['ps'].wait(timeout=5)
-            except subprocess.TimeoutExpired:
-                # If the process hasn't gracefully exited after 5 seconds we kill it
-                os.kill(pid, signal.SIGKILL)
+def kill_pg(ps, pgid, cmd):
```

Comment:
Can't you get the pgid from the ps. See comment below on putting the getpid in here

Reply:
true. will changea

Reply:
true. will change